### PR TITLE
JSON parsing: always fix all incoming json when using _manual_json

### PR DESCRIPTION
**Pull request by MarkJGx**
## Description

This change set adds non-LLM-based JSON malformity handling as a preliminary step before using the more resource-intensive LLM-based fixup.

### More Details

While running GraphRAG with a local Ollama model, I noticed frequent malformed JSON responses from LLM requests, significantly slowing down the process on an M1 Max MacBook. In a fast, parallel cloud inference system, this issue is manageable, but locally it becomes a bottleneck. After indexing, I found 140 instances of JSON parsing failures.

The `json_repair` library effectively fixed the malformed JSON in my tests. I opted not to delve into the specific parsing failure cases, as they are mainly LLM-related and predicting every edge case is impractical. This library should be robust enough to handle most local LLM faults.

## Related Issues

- https://github.com/microsoft/graphrag/issues/345

## Proposed Changes

- Add `json_repair` as a new Poetry dependency.
  - https://pypi.org/project/json-repair/
- Use `json_repair` for initial JSON repair in `_manual_json`, with a fallback to the LLM.
- Apply JSON repair when graph search JSON parsing fails.

## Checklist

- [x] Tested these changes locally.
- [x] Reviewed the code changes.
- [x] Updated documentation (if necessary).
- [ ] Added appropriate unit tests (if applicable).

### CODE REVIEW

**Comment by MarkJGx** [comment]
@microsoft-github-policy-service agree

File: graphrag/llm/openai/openai_chat_llm.py [hunk 1]
```diff
@@ -120,6 +120,7 @@ async def _manual_json(
         result = await self._invoke(input, **kwargs)
         history = result.history or []
         output = clean_up_json(result.output or "")
+        output = fix_malformed_json(result.output or "")
```
Comment (s106916) [inline graphrag/llm/openai/openai_chat_llm.py:123]: output = fix_malformed_json(output)

Comment (MarkJGx) [reply graphrag/llm/openai/openai_chat_llm.py:123]: Your comment seems to have gotten messed up, but I did notice the issue with the output not being forwarded from one function another. Just addressed.

**Comment by MarkJGx** [comment]
Hey. I've addressed the oversight caught in the comment and rebased on main, while rehashing the poetry lock with the new library added. Waiting for a review. @AlonsoGuevara @jgbradley1 et al.

**Comment by natoverse** [comment]
We have resolved several issues related to text encoding and JSON parsing that are rolled up into version [0.2.2](https://github.com/microsoft/graphrag/releases/tag/v0.2.2). Please try again with that version and re-open if this is still an issue.
